File: UI/propertease_client/lib/models/conversation.dart
import 'package:json_annotation/json_annotation.dart';
import 'application_user.dart';
import 'property.dart';

part 'conversation.g.dart';

@JsonSerializable()
class Conversation {
  int? id;
  bool? isDeleted;
  DateTime? createdAt;
  DateTime? modifiedAt;
  int? totalRecordsCount;
  Property? property;
  int? propertyId;
  ApplicationUser? client;
  int? clientId;
  ApplicationUser? renter;
  String? lastMessage;
  DateTime? lastSent;
  int? renterId;

  // Constructor
  Conversation({
    this.id = 0,
    this.isDeleted = false,
    this.createdAt,
    this.modifiedAt,
    this.totalRecordsCount = 0,
    this.property,
    this.propertyId,
    this.client,
    this.clientId,
    this.renter,
    this.renterId,
    this.lastMessage,
    this.lastSent,
  });

  // Factory constructor to create an instance from a JSON object
  factory Conversation.fromJson(Map<String, dynamic> json) =>
      _$ConversationFromJson(json);

  // Convert the instance to a JSON object
  Map<String, dynamic> toJson() => _$ConversationToJson(this);
}

File: PropertEase.Core/Entities/Conversation.cs
using PropertEase.Core.Dto.Message;
using PropertEase.Core.Entities.Base;
using PropertEase.Core.Entities.Identity;


namespace PropertEase.Core.Entities
{
    public class Conversation : BaseEntity
    {
        public Property? Property { get; set; }
        public int? PropertyId { get; set; }
        public ApplicationUser Client { get; set; }
        public int ClientId { get; set; }
        public ApplicationUser Renter { get; set; }
        public int RenterId { get; set; }
        public string? LastMessage { get; set; }
        public DateTime? LastSent { get; set; }
        public List<Message> Messages { get; set; }
    }
}
